### ISSUE TITLE
Baseline exception in selenium Java

### ISSUE THREAD

**Issue by shunguoy** [IBMa/equal-access#2077]
user reported from Slack: https://ibm-studios.slack.com/archives/C036P1CTN/p1729821107094459

I'm using the selenium java accessibility checker. I was trying to use the baselines and I'm seeing a null pointer exception. My baseline has 53 items but my actual has 60 and it's blowing up on this line with
java.lang.ArrayIndexOutOfBoundsException: Array index out of range: 53
	at com.ibm.able.equalaccess.report.BaselineManager.diff(BaselineManager.java:273)
	at com.ibm.able.equalaccess.report.BaselineManager.diffResultsWithExpected(BaselineManager.java:208)
	at com.ibm.able.equalaccess.report.BaselineManager.assertCompliance(BaselineManager.java:135)
	at com.ibm.able.equalaccess.AccessibilityChecker.assertCompliance(AccessibilityChecker.java:93)
Screenshot 2024-09-16 at 8.22.43 PM.png
 
Screenshot 2024-09-16 at 8.22.43 PM.png

**Comment by amyreit** [comment]
I could be using the calls incorrectly but if so, can you please tell me how to compare the current report to my baseline.  I was trying to do this thinking that if they didn't match, the assertCompliance call would return BASELINE_MISMATCH but stepping through the code, that's never going to work if they aren't the same size :

              
	        matchBaseline = AccessibilityChecker.assertCompliance(report);
                 if (matchBaseline==eAssertResult.BASELINE_MISMATCH) {
                       // handle a failure here
                }

I also tried this way : 

 ```
ACReport baselineRpt = AccessibilityChecker.getBaseline(pageForComplianceCheck);
	        DiffResult[] diffs = AccessibilityChecker.diffResultsWithExpected(report, baselineRpt);
```

but hits same line of code eventually and fails the same way.

Here's the code in BaselineManager.java:273. It fails here because the for loop is going from the expected length of 25 to the actual length of 35 and so right away expectedRs[25] is out of index since my expected file only has 0-24 elements. 


 ```
for (int idx=expectedRs.length; idx < actualRs.length; ++idx) {
            retVal.add(new DiffResult("A", idx, gson.toJson(expectedRs[idx]), null));  
        }
```
